kythe/typescript/indexer.ts

```diff
@@ -684,7 +706,12 @@ class Visitor {
 
     this.typeChecker = this.host.program.getTypeChecker();
 
-    this.kFile = this.newFileVName(file.fileName);
+    if (process.env.KYTHE_ROOT_DIRECTORY) {
+      this.kFile = this.newFileVName(
+          path.relative(process.env.KYTHE_ROOT_DIRECTORY, file.fileName));
```

Comment:
These environment variables are only provided during extraction, not indexing.  The indexer should limit itself to command line parameters for global options and to the specific compilation unit for compilation-unit specific things, including corpus and especially anything VName related.